src/__/collections/groupBy.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace collections;
+
+/**
+ * Returns an associative array where the keys are values of $key.
+ *
+ * Based on {@author Chauncey McAskill}'s {@link https://gist.github.com/mcaskill/baaee44487653e1afc0d array_group_by()}
+ * function.
+ *
```

Comment:
We could add usage examples here (to see usage with objects, with Closure/callable) so the doc is absolutely complete :)

Like for https://github.com/allejo/bottomline/blob/groupby/src/__/collections/get.php#L8

Reply:
For multiple usage examples, would the `**` be split or would they be together? e.g.

1.
```
/**
 *
 ** example 1
 ** // ...
 ** example 2
 ** // ..
 *
 */
```

or
 
2.
```
/**
 *
 ** example 1
 ** // ...
 *
 ** example 2
 ** // ..
 *
 */
```

Reply:
More like 2. 

There is an example there: https://github.com/maciejczyzewski/bottomline/blob/master/src/__/strings/words.php#L10)

But I know anything at PHPDoc convetions. If this is readable, for me it's good :)

Reply:
I think we just miss that to merge!

(We will handle the error management in another PR as has said Maciej)

Reply:
Oops missed my chance to add the docs. Guess it can go into a separate PR just like improved error management 👍